### PR TITLE
Guide for SSN on ANet.

### PR DESCRIPTION
**Pull request by tony2016uestc**

### CODE REVIEW

File: configs/localization/ssn/README.md [hunk 1]
```diff
@@ -37,12 +37,34 @@ You can use the following command to train a model.
...
 
+Example: train SSN on ActivityNet-1.3 dataset.
+
+1. Get the result of action proposals. (Take proposal generator BMN as an example.)
+
+```shell
+python tools/test.py configs/localization/bmn/bmn_400x100_2x8_9e_activitynet_feature.py checkpoints/SOME_CHECKPOINT.pth --eval AR@AN --out results.json
+```
+
+2. Convert the output proposal file of proposal generator (BSN, BMN) into the input proposal file of action classifier. (Currently supports SSN and P-GCN, not including TSN, I3D etc.)
```
Comment (innerlee) [inline configs/localization/ssn/README.md:54]: > ... of proposal generator (BSN, BMN)

> Currently supports SSN and P-GCN, not including TSN, I3D etc.

What's the relation between these two?

File: configs/localization/ssn/README.md [hunk 2]
```diff
@@ -37,12 +37,34 @@ You can use the following command to train a model.
...
+
+```shell
+python tools/data/activitynet/convert_proposal_format.py
+```
+
+3. train SSN model.
+
+Notes: There is a large gap between THUMOS14 and ActivityNet-1.3, therefore hyperparameters in config need to be carefully tuned.
```
Comment (innerlee) [inline configs/localization/ssn/README.md:62]: Do you have "carefully tuned" config for THUMOS14?

**Comment by innerlee** [comment]
@kennymckormick Please take a look at the changes, thanks!

File: configs/localization/ssn/README.md (hunk 2 above)
Comment (tony2016uestc) [reply configs/localization/ssn/README.md:62]: Parameters of SSN on THUMOS14 follow the setting of MMAction1, while MMAction1 does not provide the setting of SSN for ActivityNet.

File: configs/localization/ssn/README.md (hunk 1 above)
Comment (tony2016uestc) [reply configs/localization/ssn/README.md:54]: SSN and P-GCN predict categories and refine boundaries of proposals, while TSN and I3D only do classification.

Comment (innerlee) [reply configs/localization/ssn/README.md:54]: Let's try to improve the writting of this sentence.

> Convert the output proposal file of proposal generator (BSN, BMN) into the input proposal file of action classifier.

It mentions proposal generator, and BSN and BMN.

> (Currently supports SSN and P-GCN, not including TSN, I3D etc.)

Then this note mentions something else, such as SSN, P-GCN, ISN, I3D. From the perspective of an innocent reader, how are they related?

File: configs/localization/ssn/README.md (hunk 2 above)
Comment (innerlee) [reply configs/localization/ssn/README.md:62]: In L65, the config file is called `ssn_r50_1000e_anet_rgb_train.py`.
Is this new config designed for anet, or `thumos14`?

File: configs/localization/ssn/README.md [hunk 3]
```diff
@@ -37,12 +37,34 @@ You can use the following command to train a model.
...
 ```
 
-Example: train SSN model on thumos14 dataset.
+Example: train SSN on THUMOS14 dataset.
 
 ```shell
 python tools/train.py configs/localization/ssn/ssn_r50_450e_thumos14_rgb_train.py
 ```
 
+Example: train SSN on ActivityNet-1.3 dataset.
```
Comment (innerlee) [inline configs/localization/ssn/README.md:46]: Use third level header for examples.

```txt
### Example: ...
```

File: configs/localization/ssn/ssn_r50_1000e_anet_rgb_train.py [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+optimizer = dict(
+    type='SGD', lr=0.001, momentum=0.9,
+    weight_decay=1e-6)  # this lr is used for 8 gpus
+optimizer_config = dict(grad_clip=dict(max_norm=35, norm_type=2))
+# learning policy
+lr_config = dict(policy='step', step=[200, 400, 600, 800])
+checkpoint_config = dict(interval=5)
+log_config = dict(interval=1, hooks=[dict(type='TextLoggerHook')])
+# runtime settings
+total_epochs = 1000
```
Comment (kennymckormick) [inline configs/localization/ssn/ssn_r50_1000e_anet_rgb_train.py:148]: 1000 epochs seem to be a quite large number. How long does it take to train the SSN for 1000 epochs on ActivityNet?

File: configs/localization/ssn/README.md (hunk 3 above)
Comment (kennymckormick) [inline configs/localization/ssn/README.md:46]: Is there any performance number about SSN on ANet?

Comment (tony2016uestc) [reply configs/localization/ssn/README.md:46]: Not yet, I just make sure that users can train SSN with this config.

File: configs/localization/ssn/ssn_r50_1000e_anet_rgb_train.py (hunk 4 above)
Comment (tony2016uestc) [reply configs/localization/ssn/ssn_r50_1000e_anet_rgb_train.py:148]: About 2 weeks.

File: configs/localization/ssn/README.md (hunk 1 above)
Comment (tony2016uestc) [reply configs/localization/ssn/README.md:54]: The pipeline of multi-stage Temporal Action Localization:
1. Extract features with action recognizers/action classifiers (TSN, I3D etc.).
2. Generate proposals with proposal generators (BSN, BMN etc.).
3. Predict categories of proposals with action classifiers (SSN, P-GCN).

Comment (innerlee) [reply configs/localization/ssn/README.md:54]: This is much clearer now

**Comment by dreamerlin** [comment]
@tony2016uestc kindly ping :P

**Pull request closed without merging**